prow/cmd/branchprotector/protect.go

```diff
@@ -35,40 +35,31 @@ import (
 type options struct {
 	config    string
 	jobConfig string
-	token     string
 	confirm   bool
-	endpoint  flagutil.Strings
+	github    cmdutil.GitHubOptions
 }
 
 func (o *options) Validate() error {
-	if o.config == "" {
-		return errors.New("empty --config-path")
-	}
-
-	if o.token == "" {
-		return errors.New("empty --github-token-path")
+	o.github.TokenRequired = true
+	if err := o.github.Validate(!o.confirm); err != nil {
+		return err
 	}
 
-	for _, ep := range o.endpoint.Strings() {
-		_, err := url.Parse(ep)
-		if err != nil {
-			return fmt.Errorf("invalid --endpoint URL %q: %v", ep, err)
-		}
+	if o.config == "" {
+		return errors.New("empty --config-path")
 	}
 
 	return nil
 }
 
 func gatherOptions() options {
-	o := options{
-		endpoint: flagutil.NewStrings("https://api.github.com"),
-	}
-	flag.StringVar(&o.config, "config-path", "", "Path to prow config.yaml")
-	flag.StringVar(&o.jobConfig, "job-config-path", "", "Path to prow job configs.")
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.config, "config-path", "", "Path to prow config.yaml")
```

Comment:
Shouldn't the config/job config be another shared group?

Reply:
> Shouldn't the config/job config be another shared group?

I've got follow-up branches for [the config][1] and [plugin][2] agents.  But this was already pretty big, so I though review would be easier if I serialized them in follow-up PRs.  Let me know if you want them here.

[1]: https://github.com/wking/kubernetes-test-infra/tree/config-option-group
[2]: https://github.com/wking/kubernetes-test-infra/tree/plugin-option-group